test/vm.py

```diff
@@ -103,3 +106,114 @@ def run(self, cmd, user, password):
+        except ClientError as err:
+            err_code = err.response["Error"]["Code"]
+            err_msg = err.response["Error"]["Message"]
+            print(f"Couldn't create instance with image {self._ami_id} and type {self._instance_type}.")
+            print(f"Error {err_code}: {err_msg}")
+            raise
+
+    def _set_ssh_security_group(self):
+        group_name = f"bootc-image-builder-test-{str(uuid.uuid4())}"
+        group_desc = "bootc-image-builder test security group: SSH rule"
+        try:
+            print(f"Creating security group {group_name}")
+            self._ec2_security_group = self._ec2_resource.create_security_group(GroupName=group_name,
+                                                                                Description=group_desc)
+            ip_permissions = [
+                {
+                    "IpProtocol": "tcp",
+                    "FromPort": self._ssh_port,
+                    "ToPort": self._ssh_port,
+                    "IpRanges": [{"CidrIp": "0.0.0.0/0"}],
+                }
+            ]
+            print(f"Authorizing inbound rule for {group_name} ({self._ec2_security_group})")
+            self._ec2_security_group.authorize_ingress(IpPermissions=ip_permissions)
+            print("Security group created")
+        except ClientError as err:
+            err_code = err.response["Error"]["Code"]
+            err_msg = err.response["Error"]["Message"]
+            print(f"Couldn't create security group {group_name} or authorize inbound rule.")
+            print(f"Error {err_code}: {err_msg}")
+            raise
+
+    def force_stop(self):
+        if self._ec2_instance:
+            print(f"Terminating instance {self._ec2_instance.id}")
+            try:
+                self._ec2_instance.terminate()
+                self._ec2_instance.wait_until_terminated()
+                self._ec2_instance = None
+            except ClientError as err:
+                err_code = err.response["Error"]["Code"]
+                err_msg = err.response["Error"]["Message"]
+                print(f"Couldn't terminate instance {self._ec2_instance.id}.")
+                print(f"Error {err_code}: {err_msg}")
+        else:
+            print("No EC2 instance defined. Skipping termination.")
+
+        if self._ec2_security_group:
+            print(f"Deleting security group {self._ec2_security_group.id}")
+            try:
+                self._ec2_security_group.delete()
+                self._ec2_security_group = None
+            except ClientError as err:
+                err_code = err.response["Error"]["Code"]
+                err_msg = err.response["Error"]["Message"]
+                print(f"Couldn't delete security group {self._ec2_security_group.id}.")
+                print(f"Error {err_code}: {err_msg}")
+        else:
+            print("No security group defined. Skipping deletion.")
+
+    def run(self, cmd, user, password):
+        if not self._ec2_instance:
```

Comment:
if we make this `self.running()` and rename `self._ec_instance` to something like `_ip` we could share the run() code unchanged between QEMU/ec2 and move it into the VM base class, wdyt?

Reply:
I think I was expecting the `run()` function to differ more when I started and didn't revisit it after everything was done.  It makes total sense  to share it, I agree.

Reply:
Done.